tests/typescript/example.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import {StripeProvider} from '../../';
+
+// index.tsx
+const App = () => {
+  return (
+    <StripeProvider apiKey="pk_test_12345">
+      <MyStoreCheckout />
+    </StripeProvider>
+  );
+};
+
+// MyStoreCheckout.tsx
+
+import {Elements} from '../../';
+class MyStoreCheckout extends React.Component<{}, {}> {
+  render() {
+    return (
+      <Elements>
+        <CheckoutForm />
+      </Elements>
+    );
+  }
+}
+
+// CheckoutForm.tsx
+import {
+  injectStripe,
+  StripeProps,
+  CardElement,
+  CardNumberElement,
+  CardExpiryElement,
+  CardCVCElement,
+  PostalCodeElement,
+} from '../../';
+
+type BaseProps = {};
+type Props = BaseProps & StripeProps;
+
+class _CheckoutForm extends React.Component<StripeProps, {}> {
+  handleSubmit = (ev: React.FormEvent<HTMLFormElement>) => {
+    // We don't want to let default form submission happen here, which would refresh the page.
+    ev.preventDefault();
+
+    // Within the context of `Elements`, this call to createToken knows which Element to
+    // tokenize, since there's only one in this group.
+    this.props.stripe.createToken({name: 'Jenny Rosen'}).then(({token, error}) => {
+      if (token) {
+        console.log('Received Stripe token:', token.id);
+      } else if (error) {
```

Comment:
Unfortunately, you have to check both properties the way this is currently typed, since this isn't a discriminable union for typescript: https://github.com/stripe/react-stripe-elements/pull/10/files#diff-b52768974e6bc0faccb7d4b75b162c99R18